var/spack/repos/builtin/packages/sprng/package.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Sprng(AutotoolsPackage):
+    """SPRNG: A Scalable Library For Pseudorandom Number Generation
+       Sprng is a distributed process-aware random number generator that
+       avoids correlations in random number sequences across processes.
+    """
+
+    maintainers = ['kayarre']
+
+    homepage = "http://www.sprng.org"
+    url      = "http://www.sprng.org/Version5.0/sprng5.tar.bz2"
+
+    version('5.0', sha256='9172a495472cc24893e7489ce9b5654300dc60cba4430e436ce50d28eb749a66')
+
+    variant('mpi', default=True, description='Enable MPI support')
+    variant('fortran', default=False, description='Enable Fortran support')
+
+    depends_on('mpi', when='+mpi')
+
+    def url_for_version(self, version):
+        if version.version == (5, 0):
+            return "http://www.sprng.org/Version5.0/sprng5.tar.bz2"
+
+    def configure_args(self):
+        configure_args = []
+        configure_args += self.with_or_without('mpi')
+        configure_args += self.with_or_without('fortran')
+
+        if '+mpi' in self.spec:
+            mpi_link_flags = self.spec['mpi:cxx'].libs.link_flags
+            configure_args.append('LIBS={0}'.format(mpi_link_flags))
+            configure_args.append('CC={0}'.format(self.spec['mpi'].mpicc))
+            configure_args.append('CXX={0}'.format(self.spec['mpi'].mpicxx))
+            if '+fortran' in self.spec:
+                configure_args.append('FC={0}'.format(self.spec['mpi'].mpifc))
+        return configure_args
+
+    # TODO: update after solution for virtual depedencies
+    @run_before('configure')
+    def mpicxx_check(self):
+        # print(self.spec['mpi:fortran'].libs.names)
+        if '+mpi' in self.spec:
+            if 'mpi_cxx' not in self.spec['mpi:cxx'].libs.names:
+                msg = 'SPRNG requires a mpi Cxx bindings to build'
+                raise RuntimeError(msg)
+            if '+fortran' in self.spec:
+                if 'fmpi' not in self.spec['fortran'].libs.names:
+                    msg = ('SPRNG requires fortran mpi '
+                           'libraries with mpi enabled')
+                    raise RuntimeError(msg)
+        # raise RuntimeError("test")
+
+    # FIXME: update after features in #15702 are enabled
+    @run_after('build')
+    @on_package_attributes(run_tests=True)
+    def check_install(self):
```

Comment:
Should this be `@run_after('install')` or `def check_build`?

Reply:
I think this is a simple naming snafu. I don't know if it really matters when the check gets performed.
my intent was to check that it compiled after building and therefore rename `def check_install`  to `def check_build`. Is there any benefit to checking after `install` versus after `build`?

I will update it to `def check_build`